instances/models.go

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022, Unikraft GmbH and The KraftKit Authors.
+// Licensed under the BSD-3-Clause License (the "License").
+// You may not use this file except in compliance with the License.
+
+package instances
+
+import "sdk.kraft.cloud/services"
+
+// CreateRequest is the payload for a POST /instances request.
+// https://docs.kraft.cloud/api/v1/instances/#creating-a-new-instance
+type CreateRequest struct {
+	Name          string                     `json:"name"`
+	Image         string                     `json:"image"`
+	Args          []string                   `json:"args"`
+	Env           map[string]string          `json:"env"`
+	MemoryMB      int                        `json:"memory_mb"`
+	ServiceGroup  *CreateRequestServiceGroup `json:"service_group"`
+	Volumes       []CreateRequestVolume      `json:"volumes"`
+	Autostart     bool                       `json:"autostart"`
+	Replicas      int                        `json:"replicas"`
+	WaitTimeoutMs int                        `json:"wait_timeout_ms"`
+	Features      []Feature                  `json:"features"`
+}
+
+type CreateRequestServiceGroup struct {
+	UUID     *string            `json:"uuid,omitempty"`
+	Name     *string            `json:"name,omitempty"`
+	Services []services.Service `json:"services"`
+	DNSName  *string            `json:"dns_name,omitempty"`
+}
+
+type CreateRequestVolume struct {
+	UUID     string `json:"uuid"`
+	Name     string `json:"name"`
+	SizeMB   int    `json:"size_mb"`
+	At       string `json:"at"`
+	ReadOnly bool   `json:"readonly"`
+}
+
+// CreateResponseItem is a data item from a response to a POST /instances request.
+// https://docs.kraft.cloud/api/v1/instances/#creating-a-new-instance
+type CreateResponseItem struct {
+	Status      string `json:"status"`
+	State       string `json:"state"`
+	UUID        string `json:"uuid"`
+	Name        string `json:"name"`
+	FQDN        string `json:"fqdn"`
+	PrivateFQDN string `json:"private_fqdn"`
+	PrivateIP   string `json:"private_ip"`
+	BootTimeUs  int    `json:"boot_time_us"`
```

Comment:
I guess responses are not to be omitted if empty?

Reply:
There are some API responses which return `null` values, in those cases I made the fields pointers + omitempty. But I think the one you commented does not return any `null`?

Reply:
(I would be great if the documentation indicated which response attributes are optional, but I digress.)

Reply:
Yeah, I just commented on one that I saw that  it had no `omitempty`. For responses it's fine, I think things will not break.

Reply:
I performed some adjustments, applying the following rules:
  - use pointers for optional fields
  - don't bother with `omitempty` since
    - the API handles `null` values correctly in requests
    - this does not affect the deserialization of pointers in responses